azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1384,4 +1665,161 @@ private void cleanExecutionLogs() {
           - executionLogsRetentionMs);
     }
   }
-}
+
+  /*
+   * This thread is responsible for processing queued flows.
+   */
+  private class QueueProcessorThread extends Thread {
+    private static final long QUEUE_PROCESSOR_WAIT_IN_MS = 1000;
+    private static final int MAX_DISPATCHING_ERRORS_PERMITTED = 5;
+    private final long activeExecutorRefreshWindowInMilisec;
+    private final int activeExecutorRefreshWindowInFlows;
+
+    private boolean shutdown = false;
+    private boolean isActive = true;
+
+    public QueueProcessorThread(boolean isActive,
+      long activeExecutorRefreshWindowInTime,
+      int activeExecutorRefreshWindowInFlows) {
+      setActive(isActive);
+      this.activeExecutorRefreshWindowInFlows =
+        activeExecutorRefreshWindowInFlows;
+      this.activeExecutorRefreshWindowInMilisec =
+        activeExecutorRefreshWindowInTime;
+      this.setName("AzkabanWebServer-QueueProcessor-Thread");
+    }
+
+    public void setActive(boolean isActive) {
+      this.isActive = isActive;
```

Comment:
Would it be helpful for debugging purpose to log out a statement when this method is called.  This way, we can easily tell from the log.